tests/subsys/net/lib/downloader/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(downloader)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app PRIVATE ${app_sources})
+
+test_runner_generate(src/main.c)
+
+target_sources(app
+  PRIVATE
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/downloader.c
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/dl_socket.c
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/dl_parse.c
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/sanity.c
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/transports/coap.c
+  ${ZEPHYR_NRF_MODULE_DIR}/subsys/net/lib/downloader/src/transports/http.c
+)
```

Comment:
Instead of this and all the target_compile_options, is there a reason why `CONFIG_DOWNLOADER=y` in `prj.conf` doesn't work?

Reply:
Doing this works at least when socket.h is cmocked. But this is for a later phase if ever.